pkg/apis/channels/v1alpha1/subscription_validation_test.go

```diff
@@ -20,8 +20,49 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/knative/pkg/apis"
+	corev1 "k8s.io/api/core/v1"
 )
 
+const (
+	channelKind       = "Channel"
+	channelAPIVersion = "eventing.knative.dev/v1alpha1"
+	routeKind         = "Route"
+	routeAPIVersion   = "serving.knative.dev/v1alpha1"
+	FromChannelName   = "fromChannel"
+	ToChannelName     = "toChannel"
+	ProcessorName     = "processor"
+)
+
+func getValidFromRef() *corev1.ObjectReference {
```

Comment:
should also test invalid `From` and `To` ObjectReferences:
- non channel `Kind` and `APIVersion`
- fields other than `Kind`, `APIVersion` and `Name`

Reply:
Working on it... Just want to resolve the names...